vllm/envs.py

```diff
@@ -365,6 +367,18 @@ def get_default_config_root():
     (os.environ.get("VLLM_TEST_FORCE_FP8_MARLIN", "0").strip().lower() in
      ("1", "true")),
 
+    # Time in ms for the zmq client to wait for a response from the backend
+    # server for simple data operations
+    "VLLM_RPC_GET_DATA_TIMEOUT_MS":
+    lambda: int(os.getenv("VLLM_RPC_GET_DATA_TIMEOUT_MS", "1000")),
+
+    # Time in ms for the zmq client to wait for a response from the backend
+    # server for text generation.
+    # This helps ensure we don't leak open http connections and pile up
+    # deadlocked coroutines if something goes wrong with a request.
+    "VLLM_RPC_GENERATE_TIMEOUT_MS":
+    lambda: int(os.getenv("VLLM_RPC_GENERATE_TIMEOUT_MS", "300000")),
```

Comment:
@njhill thoughts on this?

I think as it stands currently this timeout only applies between each part of the generation response that is streamed back, but I know that we had wanted to maybe switch that to just one single response for unary requests.

I think in cases with super long queues this timeout could trigger as well, which may not be what we want if users did intend to just chuck a ton of requests at the server with super long timeouts

Reply:
Yes we should be conservative with this, I'm thinking no timeout at all for generate. We can/should have regular health checks.

Re the health checks, I guess now we're only doing forwarding them from the external /health endpoint. Better I think when in decoupled mode would be to have the front-end proc regularly health-check against the back-end (every few seconds), and then the front-end health check just returns whatever the latest health result was.

Reply:
👍 

Reverted generate timeout

Re: Health checks I see @robertgshaw2-neuralmagic added in the separate 10s timeout on the last PR. Seems like an okay middle ground, we can follow up with another polling loop if necessary